Dynatrace/dynatrace-operator
# Add condition for CodeModuleVersion and ProcessModuleConfig

## Description

First `Conditions` related PR, review it in a way that following `Conditions` should follow the same pattern.

Adds 2 `Conditions`:
- `CodeModuleVersion`, which is for the version-reconciler
  - A bit of a hacky solution because of the version reconciler structure, but it works 
  - Only serves as more info, not used directly for anything
- `ProcessModuleConfig` which is for the process-module-config-secret reconciler
  - Secret creation/update related, so I moved these `Conditions` into a utils package. 
  - ⚠️ Replaces the timestamp for the Dynatrace API request limit feature
     - "Basic" rule: A "transition" has (True => False, False=>True) for the timestamp to change, that is why there is an `Empty` Reason, so this transition happens, also its a nice failsafe 

## How can this be tested?

Deploy a dynakube that uses the CodeModules.
- Check the status for the `Conditions`

## Checklist
